### PR TITLE
cql_test_env: load host ID from sstables after restart

### LINKED ISSUE #14599: Microbenchmark failed on timeout
https://jenkins.scylladb.com/job/scylla-master/job/microbenchmarks/1132/ failed with the following:
```
05:54:14  WARN  2023-07-10 05:54:14,134 [shard 0] raft_group_registry - Raft server id 936f2456-ea2c-43ee-b9fd-4a3ca2491fa4 cannot be translated to an IP address.
05:59:20  WARN  2023-07-10 05:59:14,163 [shard 0] raft_group_registry - (rate limiting dropped 2996 similar messages) Raft server id 936f2456-ea2c-43ee-b9fd-4a3ca2491fa4 cannot be translated to an IP address.
06:04:25  WARN  2023-07-10 06:04:14,173 [shard 0] raft_group_registry - (rate limiting dropped 2996 similar messages) Raft server id 936f2456-ea2c-43ee-b9fd-4a3ca2491fa4 cannot be translated to an IP address.
06:09:15  WARN  2023-07-10 06:09:14,203 [shard 0] raft_group_registry - (rate limiting dropped 2996 similar messages) Raft server id 936f2456-ea2c-43ee-b9fd-4a3ca2491fa4 cannot be translated to an IP address.
06:14:20  WARN  2023-07-10 06:14:14,234 [shard 0] raft_group_registry - (rate limiting dropped 2996 similar messages) Raft server id 936f2456-ea2c-43ee-b9fd-4a3ca2491fa4 cannot be translated to an IP address.
06:19:26  WARN  2023-07-10 06:19:14,239 [shard 0] raft_group_registry - (rate limiting dropped 2996 similar messages) Raft server id 936f2456-ea2c-43ee-b9fd-4a3ca2491fa4 cannot be translated to an IP address.
06:24:16  WARN  2023-07-10 06:24:14,260 [shard 0] raft_group_registry - (rate limiting dropped 2996 similar messages) Raft server id 936f2456-ea2c-43ee-b9fd-4a3ca2491fa4 cannot be translated to an IP address.
06:29:21  WARN  2023-07-10 06:29:14,277 [shard 0] raft_group_registry - (rate limiting dropped 2996 similar messages) Raft server id 936f2456-ea2c-43ee-b9fd-4a3ca2491fa4 cannot be translated to an IP address.
06:34:26  WARN  2023-07-10 06:34:14,319 [shard 0] raft_group_registry - (rate limiting dropped 2996 similar messages) Raft server id 936f2456-ea2c-43ee-b9fd-4a3ca2491fa4 cannot be translated to an IP address.
06:39:16  WARN  2023-07-10 06:39:14,328 [shard 0] raft_group_registry - (rate limiting dropped 2996 similar messages) Raft server id 936f2456-ea2c-43ee-b9fd-4a3ca2491fa4 cannot be translated to an IP address.
06:44:21  WARN  2023-07-10 06:44:14,352 [shard 0] raft_group_registry - (rate limiting dropped 2996 similar messages) Raft server id 936f2456-ea2c-43ee-b9fd-4a3ca2491fa4 cannot be translated to an IP address.
06:49:26  WARN  2023-07-10 06:49:14,367 [shard 0] raft_group_registry - (rate limiting dropped 2996 similar messages) Raft server id 936f2456-ea2c-43ee-b9fd-4a3ca2491fa4 cannot be translated to an IP address.
06:54:17  WARN  2023-07-10 06:54:14,380 [shard 0] raft_group_registry - (rate limiting dropped 2996 similar messages) Raft server id 936f2456-ea2c-43ee-b9fd-4a3ca2491fa4 cannot be translated to an IP address.
06:59:22  WARN  2023-07-10 06:59:14,401 [shard 0] raft_group_registry - (rate limiting dropped 2996 similar messages) Raft server id 936f2456-ea2c-43ee-b9fd-4a3ca2491fa4 cannot be translated to an IP address.
07:04:27  WARN  2023-07-10 07:04:14,413 [shard 0] raft_group_registry - (rate limiting dropped 2996 similar messages) Raft server id 936f2456-ea2c-43ee-b9fd-4a3ca2491fa4 cannot be translated to an IP address.
07:09:18  WARN  2023-07-10 07:09:14,431 [shard 0] raft_group_registry - (rate limiting dropped 2996 similar messages) Raft server id 936f2456-ea2c-43ee-b9fd-4a3ca2491fa4 cannot be translated to an IP address.
```

Based on Slack chat with @kbr-
```
[2023-07-10T02:54:14.484Z] WARN  2023-07-10 05:54:14,092 [shard 0] sstable - Host id 936f2456-ea2c-43ee-b9fd-4a3ca2491fa4 does not match local host id b25cb53c-7863-460a-95cc-512a4df0b6a7 while validating SSTable: ./perf_large_partition_data/system/raft_snapshots-6fa537a77e97344e9034b69c7ef69be6/me-3g7j_082d_1qssw2jfu9y44i2zue-big-Data.db. Load foreign SSTables via the upload dir instead.
[2023-07-10T02:54:14.484Z] WARN  2023-07-10 05:54:14,093 [shard 0] sstable - Host id 936f2456-ea2c-43ee-b9fd-4a3ca2491fa4 does not match local host id b25cb53c-7863-460a-95cc-512a4df0b6a7 while validating SSTable: ./perf_large_partition_data/system/raft_snapshots-6fa537a77e97344e9034b69c7ef69be6/me-3g7j_082d_1qssw2jfu9y44i2zue-big-Data.db. Load foreign SSTables via the upload dir instead.
[2023-07-10T02:54:14.484Z] WARN  2023-07-10 05:54:14,109 [shard 0] database - Skipping undefined keyspace: commitlog.dir
[2023-07-10T02:54:14.484Z] WARN  2023-07-10 05:54:14,109 [shard 0] database - Skipping undefined keyspace: hints.dir
[2023-07-10T02:54:14.484Z] WARN  2023-07-10 05:54:14,109 [shard 0] database - Skipping undefined keyspace: view_hints.dir
[2023-07-10T02:54:14.485Z] WARN  2023-07-10 05:54:14,116 [shard 0] sstable - Host id 936f2456-ea2c-43ee-b9fd-4a3ca2491fa4 does not match local host id b25cb53c-7863-460a-95cc-512a4df0b6a7 while validating SSTable: ./perf_large_partition_data/system_distributed/cdc_streams_descriptions_v2-0bf73fd765b236b085e5658131d5df36/me-3g7j_082d_1nss12jfu9y44i2zue-big-Data.db. Load foreign SSTables via the upload dir instead.
[2023-07-10T02:54:14.485Z] WARN  2023-07-10 05:54:14,116 [shard 0] sstable - Host id 936f2456-ea2c-43ee-b9fd-4a3ca2491fa4 does not match local host id b25cb53c-7863-460a-95cc-512a4df0b6a7 while validating SSTable: ./perf_large_partition_data/system_distributed/cdc_generation_timestamps-fdf455c4cfec3e009719d7a45436c89d/me-3g7j_082d_1pxxs2jfu9y44i2zue-big-Data.db. Load foreign SSTables via the upload dir instead.
[2023-07-10T02:54:14.485Z] WARN  2023-07-10 05:54:14,116 [shard 0] sstable - Host id 936f2456-ea2c-43ee-b9fd-4a3ca2491fa4 does not match local host id b25cb53c-7863-460a-95cc-512a4df0b6a7 while validating SSTable: ./perf_large_partition_data/system_distributed_everywhere/cdc_generation_descriptions_v2-234d2227dd633d37ac5fc013e2ea9e6e/me-3g7j_082d_1ofxc2jfu9y44i2zue-big-Data.db. Load foreign SSTables via the upload dir instead.
[2023-07-10T02:54:14.485Z] WARN  2023-07-10 05:54:14,116 [shard 0] sstable - Host id 936f2456-ea2c-43ee-b9fd-4a3ca2491fa4 does not match local host id b25cb53c-7863-460a-95cc-512a4df0b6a7 while validating SSTable: ./perf_large_partition_data/system_auth/roles-5bc52802de2535edaeab188eecebb090/me-3g7j_082d_1aq3k2jfu9y44i2zue-big-Data.db. Load foreign SSTables via the upload dir instead.
[2023-07-10T02:54:14.485Z] WARN  2023-07-10 05:54:14,118 [shard 0] sstable - Host id 936f2456-ea2c-43ee-b9fd-4a3ca2491fa4 does not match local host id b25cb53c-7863-460a-95cc-512a4df0b6a7 while validating SSTable: ./perf_large_partition_data/system_distributed/cdc_streams_descriptions_v2-0bf73fd765b236b085e5658131d5df36/me-3g7j_082d_1nss12jfu9y44i2zue-big-Data.db. Load foreign SSTables via the upload dir instead.
[2023-07-10T02:54:14.485Z] WARN  2023-07-10 05:54:14,118 [shard 0] sstable - Host id 936f2456-ea2c-43ee-b9fd-4a3ca2491fa4 does not match local host id b25cb53c-7863-460a-95cc-512a4df0b6a7 while validating SSTable: ./perf_large_partition_data/system_distributed/cdc_generation_timestamps-fdf455c4cfec3e009719d7a45436c89d/me-3g7j_082d_1pxxs2jfu9y44i2zue-big-Data.db. Load foreign SSTables via the upload dir instead.
[2023-07-10T02:54:14.485Z] WARN  2023-07-10 05:54:14,118 [shard 0] sstable - Host id 936f2456-ea2c-43ee-b9fd-4a3ca2491fa4 does not match local host id b25cb53c-7863-460a-95cc-512a4df0b6a7 while validating SSTable: ./perf_large_partition_data/system_distributed_everywhere/cdc_generation_descriptions_v2-234d2227dd633d37ac5fc013e2ea9e6e/me-3g7j_082d_1ofxc2jfu9y44i2zue-big-Data.db. Load foreign SSTables via the upload dir instead.
[2023-07-10T02:54:14.485Z] WARN  2023-07-10 05:54:14,118 [shard 0] sstable - Host id 936f2456-ea2c-43ee-b9fd-4a3ca2491fa4 does not match local host id b25cb53c-7863-460a-95cc-512a4df0b6a7 while validating SSTable: ./perf_large_partition_data/system_auth/roles-5bc52802de2535edaeab188eecebb090/me-3g7j_082d_1aq3k2jfu9y44i2zue-big-Data.db. Load foreign SSTables via the upload dir instead.
[2023-07-10T02:54:14.485Z] WARN  2023-07-10 05:54:14,119 [shard 0] sstable - Host id 936f2456-ea2c-43ee-b9fd-4a3ca2491fa4 does not match local host id b25cb53c-7863-460a-95cc-512a4df0b6a7 while validating SSTable: ./perf_large_partition_data/ks/config-0055b5501ecd11eea706a8782cebc3f6/me-3g7j_082d_1a2y92jfu9y44i2zue-big-Data.db. Load foreign SSTables via the upload dir instead.
[2023-07-10T02:54:14.485Z] WARN  2023-07-10 05:54:14,119 [shard 0] sstable - Host id 936f2456-ea2c-43ee-b9fd-4a3ca2491fa4 does not match local host id b25cb53c-7863-460a-95cc-512a4df0b6a7 while validating SSTable: ./perf_large_partition_data/ks/large_part_ds1-0057b1201ecd11eea706a8782cebc3f6/me-3g7j_081x_4lw812jfu9y44i2zue-big-Data.db. Load foreign SSTables via the upload dir instead.
[2023-07-10T02:54:14.486Z] WARN  2023-07-10 05:54:14,119 [shard 0] sstable - Host id 936f2456-ea2c-43ee-b9fd-4a3ca2491fa4 does not match local host id b25cb53c-7863-460a-95cc-512a4df0b6a7 while validating SSTable: ./perf_large_partition_data/ks/small_part-04c4cf401ecd11eea706a8782cebc3f6/me-3g7j_0828_0eseo2jfu9y44i2zue-big-Data.db. Load foreign SSTables via the upload dir instead.
[2023-07-10T02:54:14.486Z] WARN  2023-07-10 05:54:14,122 [shard 0] sstable - Host id 936f2456-ea2c-43ee-b9fd-4a3ca2491fa4 does not match local host id b25cb53c-7863-460a-95cc-512a4df0b6a7 while validating SSTable: ./perf_large_partition_data/ks/config-0055b5501ecd11eea706a8782cebc3f6/me-3g7j_082d_1a2y92jfu9y44i2zue-big-Data.db. Load foreign SSTables via the upload dir instead.
[2023-07-10T02:54:14.486Z] WARN  2023-07-10 05:54:14,122 [shard 0] sstable - Host id 936f2456-ea2c-43ee-b9fd-4a3ca2491fa4 does not match local host id b25cb53c-7863-460a-95cc-512a4df0b6a7 while validating SSTable: ./perf_large_partition_data/ks/large_part_ds1-0057b1201ecd11eea706a8782cebc3f6/me-3g7j_081x_4lw812jfu9y44i2zue-big-Data.db. Load foreign SSTables via the upload dir instead.
[2023-07-10T02:54:14.486Z] WARN  2023-07-10 05:54:14,122 [shard 0] sstable - Host id 936f2456-ea2c-43ee-b9fd-4a3ca2491fa4 does not match local host id b25cb53c-7863-460a-95cc-512a4df0b6a7 while validating SSTable: ./perf_large_partition_data/ks/small_part-04c4cf401ecd11eea706a8782cebc3f6/me-3g7j_0828_0eseo2jfu9y44i2zue-big-Data.db. Load foreign SSTables via the upload dir instead.
```
@kbr-  wrote:
```
looks like the node changed its Host ID (which should never happen).
and 936f2456-ea2c-43ee-b9fd-4a3ca2491fa4 looks like the node's old host ID based on the sstable read errors
it looks like the node failed to load its system.local sstable which caused it to generate a new host ID
```

### PR DESCRIPTION
**Pull request by kbr-scylla**
Performance tests such as `perf-fast-forward` are executed in our CI environments in two steps (two invocations of the `scylla` process): first by populating data directories (with `--populate` option), then by running the actual test.

These tests are using `cql_test_env`, which did not load the previously saved (in the populate step) Host ID of this node, but generated a new one randomly instead.

In b39ca9791911165c2ad2de09ca79adcd366f7061 we enabled `consistent_cluster_management` by default. This caused the perf tests to hang in `setup_group0` at `read_barrier` step. That's because Raft group 0 was initialized with old configuration -- the one created during the populate step -- but the Raft server was started with a newly generated Host ID (which is used as the server's Raft ID), so the server considered itself as being outside the configuration.

Fix this by reloading the Host ID from disk, simulating more closely the behavior of main.cc initialization.

Fixes #14599

### CODE REVIEW

**Comment by kbr-scylla** [comment]
Ref for the FIXME I added: #14638

File: test/lib/cql_test_env.cc [hunk 1]
```diff
@@ -591,7 +588,9 @@ class single_node_cql_env : public cql_test_env {
 
             sharded<locator::shared_token_metadata> token_metadata;
             locator::token_metadata::config tm_cfg;
-            tm_cfg.topo_cfg.this_host_id = cfg->host_id;
+            // FIXME: should we update this later? #14638
+            // main.cc doesn't update it either, we simulate the behavior.
```
Comment (bhalevy) [inline test/lib/cql_test_env.cc:592]: It is used only by unit tests.
It's more than ok for cql_test_env to mimic main

Comment (bhalevy) [reply test/lib/cql_test_env.cc:592]: I don't think this deserves a FIXME.
A comment explaining that cql_test_env is doing the same as main would be enough

**Comment by kbr-scylla** [comment]
> CI state ABORTED - https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/2401/

spot termination.

v2: removed `FIXME`, simplified the comment.

**Review by bhalevy — APPROVED** [review]

**Comment by avikivity** [comment]
I would have preferred for cql_test_env to have a fixed id, but this is okay too.

**Pull request closed without merging**